### ISSUE TITLE
Using regression loss for multi-class classification

### ISSUE THREAD

**Issue by mmahmedgithub** [chemprop/chemprop#839]
Hello,

I am wondering how relevant is using regression loss functions (e.g. RMSE) for multi-class classification modelling in Chemprop. I have just read this Manucript and they claim that this is recommended. I am wondering if Chemprop developers have done any experiments for that:

**Manucript**:
https://slas-discovery.org/article/S2472-5552(24)00006-6/fulltext

Look at: 2.7 Statistical parameters

Thanks,
Marawan

**Comment by JacksonBurns** [comment]
The paper you referenced sorted all of their targets into bins and then labeled those bins 0, 1, and 2. The metric which was being used for the Kaggle competition that provided that data was not available in the software the authors referenced, so they simply used RMSE.

Neither Chemprop v1 nor v2 implement the QWK method, so you are free to do the same thing as the referenced authors and just use RMSE instead or implement QWK yourself.

Whether or not this is a _good_ idea is a 'research question' which we can't definitively answer. You can try (1) training a regression model directly (2) training on binned data as classification or (3) training on binned data as regression. Whichever one works best is the one you should go with - just keep in mind that the data may be imbalanced, binning can make this better or worse.

**Comment by davidegraff** [comment]
To add some more context, framing classification as a regression task with labels $k \in [K]$, where $K$ is the number of classes and $[K] = [1 \dots K]$ and minimizing MSE is called the [Brier score](https://en.wikipedia.org/wiki/Brier_score). It's an empiricism deciding whether minimizing cross entropy or the Brier score is better, but history has clearly favored one over the other so you can probably guess what performance to expect.
